leshan-server-core/src/main/java/org/eclipse/leshan/server/queue/impl/ResponseProcessingTask.java

```diff
@@ -0,0 +1,102 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Bosch Software Innovations GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ *
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ *
+ * Contributors:
+ *     Alexander Ellwein, Daniel Maier (Bosch Software Innovations GmbH)
+ *                                - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.leshan.server.queue.impl;
+
+import org.eclipse.leshan.core.response.ErrorCallback;
+import org.eclipse.leshan.core.response.LwM2mResponse;
+import org.eclipse.leshan.core.response.ResponseCallback;
+import org.eclipse.leshan.server.queue.QueuedRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+/**
+ * A response processing task is responsible for calling back the registered listeners with response or error result.
+ *
+ */
+class ResponseProcessingTask implements Runnable {
+
+    private final Logger LOG = LoggerFactory.getLogger(ResponseProcessingTask.class);
+
+    private final Exception exception;
+    private final boolean hasException;
+    private final LwM2mResponse response;
+    private final Map<Long, ResponseContext> responseContext;
+
+    private final QueuedRequest request;
+
+    /**
+     * Creates a new task for processing response on exception.
+     *
+     * @param request request being processed
+     * @param responseContext response context map for mapping response ID to the callback
+     * @param exception exception to propagate
+     */
+    ResponseProcessingTask(QueuedRequest request, Map<Long, ResponseContext> responseContext,
```

Comment:
Yes, we need just the response context here. But later when `run()` is complete, the response context for this request should be cleaned up.